pkg/collector/process.go

```diff
@@ -0,0 +1,266 @@
+// Copyright 2020 VMware, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collector
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"net"
+	"sync"
+	"time"
+
+	"k8s.io/klog/v2"
+
+	"github.com/vmware/go-ipfix/pkg/config"
+	"github.com/vmware/go-ipfix/pkg/entities"
+)
+
+type collectingProcess struct {
+	// for each obsDomainID, there is a map of templates
+	templatesMap map[uint32]map[uint16][]*templateField
+	// templatesLock allows multiple readers or one writer at the same time
+	templatesLock sync.RWMutex
+	// template lifetime
+	templateTTL uint32
+	// server information
+	address net.Addr
+	// maximum buffer size to read the record
+	maxBufferSize uint16
+	// chanel to receive stop information
+	stopChan chan bool
+	// packet list
+	messages []*entities.Message
+	// channels to handle clients
+	clients map[string]*client
+}
+
+type templateField struct {
+	elementID     uint16
+	elementLength uint16
+	enterpriseID  uint32
+}
+
+type client struct {
```

Comment:
Prefer clientHandler as all fields in the struct is used for handling.

Reply:
Agreed.